### DISCUSSION TITLE
Why are many of the files in parcel's internals JavaScript?

### DISCUSSION THREAD

**Discussion by tristan-f-r** [parcel-bundler/parcel#8729] [Q&A] [answered]
I was going to try my hand at #7503, and when I began working on it with VSCode, I noticed that despite the usage of types and TypeScript syntax, a lot of the internal script files are JavaScript instead of TypeScript. So, my two questions are:

1. Why is this the case? (I can only assume it's for a good reason)
2. How can I configure my editor to recognize JavaScript files as TypeScript in the future? (And possibly offer a note in [CONTRIBUTING.md](https://github.com/parcel-bundler/parcel/blob/v2/CONTRIBUTING.md) for others to see.)

**Comment by devongovett** [comment, marked as answer]
We use [Flow](https://flow.org) rather than TypeScript. You can install the [Flow for VSCode extension](https://marketplace.visualstudio.com/items?itemName=flowtype.flow-for-vscode) to get type checking and autocomplete working as expected (see linked website for instructions).
